feat: integrating new log sync

benesjan opened the pull request:
Fixes #17775

In this PR I integrate the new recipient log sync algo  and the old one gets dropped. The file structure is still a bit messy but that I will handle in a followup PR to make this PR not insane to review.

I Introduced `SenderAddressBook` that was originally part of `RecipientTaggingDataProvider`. I decided to do this separation because I want the new log sync algo to not be aware of sender addresses etc. - I wanted it to deal only with `DirectionalAppTaggingSecret`s as it feels like a nice separation.

benesjan commented:
* **#19290** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19290?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19289** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19289?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19233** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19233?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19125** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19125?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19125?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#19161** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19161?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19158** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19158?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19157** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19157?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19030** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19030?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17611** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17611?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17568** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17568?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17534** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17534?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17445** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17445?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17451** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17451?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/pxe/src/logs/log_service.test.ts:
```diff
@@ -1,437 +1,60 @@
+  let sendersDataProvider: SendersDataProvider;
   let logService: LogService;
 
-  describe('sync tagged logs', () => {
```
These were the old algorithm tests. The new algo tests are in `yarn-project/pxe/src/tagging/recipient_sync/` directory.

I didn't modify the `bulkRetrieveLogs` tests here - I just needed to update its test setup because it was messed up a bit and it relied on the test setup of the "sync tagged logs" tests.

benesjan commented on noir-projects/noir-contracts/contracts/test/pending_note_hashes_contract/src/main.nr:
```diff
@@ -294,50 +294,66 @@ pub contract PendingNoteHashes {
     //}
 
     #[external("private")]
```
This is used by the `e2e_pending_note_hashes_contract` test and it became more convoluted because of [this](https://aztecprotocol.slack.com/archives/C07R3GTKJ49/p1767158343532519?thread_ts=1766541757.471949&cid=C07R3GTKJ49). Now I need to pass in recipients in order to not send all the notes to one recipient - this would overflow `UNFINALIZED_TAGGING_INDEXES_WINDOW_LEN` which got dropped to 20 from 95.

benesjan commented on yarn-project/pxe/src/storage/tagging_data_provider/sender_address_book.ts:
```diff
@@ -0,0 +1,42 @@
+import { toArray } from '@aztec/foundation/iterable';
```
This was originally part of the same data provider (of the recipient tagging data provider). I think it's clearer to have it separate as the recipient log sync is now not aware of senders - it just receives directional app tagging secrets on the input.

benesjan commented on yarn-project/pxe/src/tagging/recipient_sync/recipient_tagging_data_provider.ts:
```diff
@@ -9,9 +9,9 @@ import type { DirectionalAppTaggingSecret } from '@aztec/stdlib/logs';
  * blocks, which by definition cannot be affected by reorgs.
  *
- * TODO(benesjan): Rename as to RecipientTaggingDataProvider and relocate once the old tagging sync is purged.
+ * TODO(benesjan): Relocate to yarn-project/pxe/src/storage/tagging_data_provider
```
This will be done in a followup PR.

benesjan commented on noir-projects/noir-contracts/contracts/test/pending_note_hashes_contract/src/main.nr:
```diff
@@ -294,50 +294,66 @@ pub contract PendingNoteHashes {
     #[external("private")]
-    fn test_recursively_create_notes(
-        owner: AztecAddress,
-        sender: AztecAddress,
```
The sender arg was unused so I dropped it here - messy code.

mverzilli commented on yarn-project/pxe/src/logs/log_service.test.ts:
+18 -395 line diffs is music to my ears ❤️

mverzilli commented on yarn-project/pxe/src/logs/log_service.ts:
just double checking if this comment is still in sync with reality

benesjan replied:
This comment was indeed stale. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/19125/commits/170490cb2372f5b79ffce3921001e7d6f8cfabbe

Thanks

mverzilli commented on yarn-project/pxe/src/logs/log_service.ts:
```diff
@@ -140,165 +144,58 @@ export class LogService {
-      ...(await this.keyStore.getAccounts()),
-    ].filter((address, index, self) => index === self.findIndex(otherAddress => otherAddress.equals(address)));
-    const secrets = await Promise.all(
+    const senders = [...(await this.senderAddressBook.getSenders()), ...(await this.keyStore.getAccounts())].filter(
```
Suggestion:

```ts
const [addressBookSenders, keyStoreAccountAddresses] = await Promise.all([
  this.senderAddressBook.getSenders(),
  this.keyStore.getAccounts()
];
const senders = [...addressBookSenders, ...keyStoreAccountAddresses].filter(etc...);
```

looks a bit cleaner IMO

benesjan replied:
I found the filtration logic there confusing so I decided to use a `Set` instead: https://github.com/AztecProtocol/aztec-packages/pull/19125/commits/170490cb2372f5b79ffce3921001e7d6f8cfabbe

mverzilli commented on yarn-project/pxe/src/storage/tagging_data_provider/sender_address_book.ts:
```diff
@@ -0,0 +1,42 @@
+ * Stores sender addresses. During recipient log synchronization, these senders are used, along with a given recipient,
+ * to derive directional app tagging secrets that are then used to sync the logs.
+ */
+export class SenderAddressBook {
```
I really like the `SenderAddressBook` name but I'm a bit anxious about breaking the `*DataProvider` convention: it might obscure the fact that this is another `DataProvider` and reduce "grepability" of the codebase.

Sidenote: Maybe `DataProvider` is not the happiest name, and we could move to calling them `Store`s (so `SenderAddressBookStore` is much happier than an eventually really ugly and long `SenderAddressBookDataProvider`)

Thoughts?

benesjan replied:
Agree with renaming it all to a `Store` in a followup PR 👍

benesjan replied:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/19290

mverzilli reviewed (approved):
Nice job! Only some nits to discuss but they don't need to block the PR

benesjan committed:
feat: integrating new log sync

Fixes #17775

In this PR I integrate the new recipient log sync algo  and the old one gets dropped. The file structure is still a bit messy but that I will handle in a followup PR to make this PR not insane to review.

I Introduced `SenderAddressBook` that was originally part of `RecipientTaggingDataProvider`. I decided to do this separation because I want the new log sync algo to not be aware of sender addresses etc. - I wanted it to deal only with `DirectionalAppTaggingSecret`s as it feels like a nice separation.
